INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [11027] using StatReload
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/_subprocess.py", line 80, in subprocess_started
    target(sockets=sockets)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 70, in serve
    await self._serve(sockets)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 77, in _serve
    config.load()
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/config.py", line 435, in load
    self.loaded_app = import_from_string(self.app)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 22, in import_from_string
    raise exc from None
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend/app/main.py", line 11, in <module>
    from .auth import (
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend/app/auth.py", line 4, in <module>
    from passlib.context import CryptContext
ModuleNotFoundError: No module named 'passlib'
INFO:     Stopping reloader process [11027]
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [11371] using StatReload
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/_subprocess.py", line 80, in subprocess_started
    target(sockets=sockets)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 70, in serve
    await self._serve(sockets)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/server.py", line 77, in _serve
    config.load()
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/config.py", line 435, in load
    self.loaded_app = import_from_string(self.app)
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 22, in import_from_string
    raise exc from None
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend/app/main.py", line 11, in <module>
    from .auth import (
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend/app/auth.py", line 4, in <module>
    from passlib.context import CryptContext
ModuleNotFoundError: No module named 'passlib'
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
ERROR:    [Errno 48] Address already in use
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [13671] using StatReload
INFO:     Started server process [13739]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:53845 - "OPTIONS /token HTTP/1.1" 200 OK
(trapped) error reading bcrypt version
Traceback (most recent call last):
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/passlib/handlers/bcrypt.py", line 620, in _load_backend_mixin
    version = _bcrypt.__about__.__version__
AttributeError: module 'bcrypt' has no attribute '__about__'
INFO:     127.0.0.1:53845 - "POST /token HTTP/1.1" 200 OK
INFO:     127.0.0.1:53845 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53847 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "OPTIONS /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:53885 - "OPTIONS /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:53849 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:53887 - "OPTIONS /routes HTTP/1.1" 200 OK
Making OSRM request to: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.008200,30.282100 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Making OSRM request to: https://router.project-osrm.org/route/v1/car/78.037608,30.287645;78.008200,30.282100 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Trying waypoint route: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.027904,30.289872;78.008200,30.282100
INFO:     127.0.0.1:53887 - "POST /routes HTTP/1.1" 200 OK
Making OSRM request to: https://router.project-osrm.org/route/v1/driving/78.008200,30.282100;78.008200,30.282100 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Making OSRM request to: https://router.project-osrm.org/route/v1/car/78.008200,30.282100;78.008200,30.282100 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Trying waypoint route: https://router.project-osrm.org/route/v1/driving/78.008200,30.282100;78.013200,30.287100;78.008200,30.282100
INFO:     127.0.0.1:53898 - "POST /routes HTTP/1.1" 500 Internal Server Error
Making OSRM request to: https://router.project-osrm.org/route/v1/driving/78.008200,30.282100;78.066221,30.302542 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Got 2 alternative routes
INFO:     127.0.0.1:53911 - "POST /routes HTTP/1.1" 200 OK
Making OSRM request to: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.079261,30.372744 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Making OSRM request to: https://router.project-osrm.org/route/v1/car/78.037608,30.287645;78.079261,30.372744 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Trying waypoint route: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.063434,30.335195;78.079261,30.372744
INFO:     127.0.0.1:53915 - "POST /routes HTTP/1.1" 200 OK
Making OSRM request to: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.079261,30.372744 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Making OSRM request to: https://router.project-osrm.org/route/v1/car/78.037608,30.287645;78.079261,30.372744 with params: {'overview': 'full', 'geometries': 'geojson'}
OSRM request successful, got 1 routes
Trying waypoint route: https://router.project-osrm.org/route/v1/driving/78.037608,30.287645;78.063434,30.335195;78.079261,30.372744
INFO:     127.0.0.1:53922 - "POST /routes HTTP/1.1" 200 OK
INFO:     127.0.0.1:53956 - "GET /locations HTTP/1.1" 200 OK
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [13739]
INFO:     Started server process [14444]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
ERROR:    [Errno 48] Address already in use
INFO:     127.0.0.1:54042 - "OPTIONS /token HTTP/1.1" 200 OK
(trapped) error reading bcrypt version
Traceback (most recent call last):
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/passlib/handlers/bcrypt.py", line 620, in _load_backend_mixin
    version = _bcrypt.__about__.__version__
AttributeError: module 'bcrypt' has no attribute '__about__'
INFO:     127.0.0.1:54042 - "POST /token HTTP/1.1" 200 OK
INFO:     127.0.0.1:54042 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:54044 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:54042 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:54044 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:54046 - "OPTIONS /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:54048 - "OPTIONS /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:54046 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:54046 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:54050 - "OPTIONS /routes HTTP/1.1" 200 OK
ORS error: 400 {"error":{"code":2012,"message":"Unknown parameter 'alternatives'."},"info":{"engine":{"build_date":"2025-05-06T08:31:01Z","graph_version":"1","graph_date":"2025-05-18T13:01:56Z","version":"9.2.0"},"timestamp":1747649529577}}
INFO:     127.0.0.1:54050 - "POST /routes HTTP/1.1" 500 Internal Server Error
ORS error: 400 {"error":{"code":2012,"message":"Unknown parameter 'alternatives'."},"info":{"engine":{"build_date":"2025-05-06T08:31:01Z","graph_version":"1","graph_date":"2025-05-18T13:01:56Z","version":"9.2.0"},"timestamp":1747649544968}}
INFO:     127.0.0.1:54052 - "POST /routes HTTP/1.1" 500 Internal Server Error
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [14444]
INFO:     Started server process [15046]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
ERROR:    [Errno 48] Address already in use
INFO:     127.0.0.1:54476 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:54476 - "GET /locations HTTP/1.1" 200 OK
ORS error: 400 {"error":{"code":2012,"message":"Unknown parameter 'alternatives'."},"info":{"engine":{"build_date":"2025-05-06T08:31:01Z","graph_version":"1","graph_date":"2025-05-18T13:01:56Z","version":"9.2.0"},"timestamp":1747649649649}}
WARNING: Using fallback direct path for ISBT Dehradun to Kargi Chowk as OSRM failed
INFO:     127.0.0.1:54528 - "POST /routes HTTP/1.1" 200 OK
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [15046]
INFO:     Started server process [15620]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [15620]
INFO:     Started server process [15968]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Will watch for changes in these directories: ['/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/backend']
ERROR:    [Errno 48] Address already in use
INFO:     127.0.0.1:56108 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:56108 - "GET /locations HTTP/1.1" 200 OK
INFO:     127.0.0.1:56154 - "POST /routes HTTP/1.1" 200 OK
INFO:     127.0.0.1:56332 - "POST /routes HTTP/1.1" 200 OK
INFO:     127.0.0.1:56410 - "POST /routes HTTP/1.1" 200 OK
INFO:     127.0.0.1:56489 - "POST /routes HTTP/1.1" 200 OK
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [15968]
INFO:     Started server process [17121]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:60639 - "OPTIONS /token HTTP/1.1" 200 OK
(trapped) error reading bcrypt version
Traceback (most recent call last):
  File "/Users/alok/Documents/Daa pbl/optimal_route_finder_v2/venv/lib/python3.10/site-packages/passlib/handlers/bcrypt.py", line 620, in _load_backend_mixin
    version = _bcrypt.__about__.__version__
AttributeError: module 'bcrypt' has no attribute '__about__'
INFO:     127.0.0.1:60639 - "POST /token HTTP/1.1" 200 OK
INFO:     127.0.0.1:60639 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60641 - "OPTIONS /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60639 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60641 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60641 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60641 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
INFO:     127.0.0.1:60647 - "GET /routes/history HTTP/1.1" 200 OK
WARNING:  StatReload detected changes in 'app/route_service.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [17121]
INFO:     Started server process [19551]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
WARNING:  StatReload detected changes in 'app/main.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [19551]
INFO:     Started server process [19563]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [19563]
INFO:     Stopping reloader process [13671]
